source/frontend/src/features/mapSideBar/compensation/detail/CompensationRequisitionHistoricalDetailContainer.tsx

```diff
@@ -0,0 +1,350 @@
+    getCompensationRequisitionAtTime: {
+      execute: getCompensationRequisition,
+      loading: getCompensationRequisitionLoading,
+    },
+    getCompensationRequisitionAcqPropertiesAtTime: {
+      execute: getCompensationAcqProperties,
+      loading: loadingCompReqAcqProperties,
+    },
+    getCompensationRequisitionLeasePropertiesAtTime: {
+      execute: getCompensationLeaseProperties,
+      loading: loadingCompReqLeaseProperties,
+    },
+    getCompensationRequisitionAcqPayeesAtTime: {
+      execute: getCompensationAcqPayees,
+      loading: loadingCompReqAcqPayees,
+    },
+    getCompensationRequisitionLeasePayeesAtTime: {
+      execute: getCompensationLeasePayees,
+      loading: loadingCompReqLeasePayees,
+    },
+  } = useCompensationRequisitionRepository();
+
+  const {
+    getAcquisitionAtTime: { execute: getAcquisition },
+  } = useAcquisitionProvider();
+
+  const {
+    getLeaseAtTime: { execute: getLease },
+  } = useLeaseRepository();
+
+  const {
+    getProjectAtTime: { execute: getProject },
+  } = useProjectProvider();
+
+  const {
+    getProductAtTime: { execute: getProduct },
+  } = useProductProvider();
+
+  const fetchParentFile = useCallback(async () => {
+    if (fileType === ApiGen_CodeTypes_FileTypes.Acquisition) {
+      const response = await getAcquisition(parentFileId, adjustedTime);
+      setParentFile(response);
+    } else if (fileType === ApiGen_CodeTypes_FileTypes.Lease) {
+      const response = await getLease(parentFileId, adjustedTime);
+      setParentFile(response);
+    }
+  }, [parentFileId, fileType, adjustedTime, getAcquisition, getLease]);
+
+  const fetchProject = useCallback(
+    async (
+      projectId: number,
+      setter: (value: React.SetStateAction<ApiGen_Concepts_Project>) => void,
+    ) => {
+      const response = await getProject(projectId, adjustedTime);
+      setter(response);
+    },
+    [adjustedTime, getProject],
+  );
+
+  const fetchProduct = useCallback(
+    async (product: number) => {
+      setProduct(null);
+      const response = await getProduct(product, adjustedTime);
+      setProduct(response);
+    },
+    [adjustedTime, getProduct],
+  );
+
+  const fetchCompensationRequisition = useCallback(async () => {
+    setTimedCompensation(null);
+    const response = await getCompensationRequisition(compensationRequisitionId, adjustedTime);
+    setTimedCompensation(response);
+  }, [compensationRequisitionId, adjustedTime, getCompensationRequisition]);
+
+  const fetchCompensationProperties = useCallback(async () => {
+    if (fileType === ApiGen_CodeTypes_FileTypes.Acquisition) {
+      const compReqProperties = await getCompensationAcqProperties(
+        compensationRequisitionId,
+        adjustedTime,
+      );
+      setCompensationRequisitionProperties(compReqProperties ?? []);
+    }
+    if (fileType === ApiGen_CodeTypes_FileTypes.Lease) {
+      const compReqProperties = await getCompensationLeaseProperties(
+        compensationRequisitionId,
```

Comment:
I think at this point we can assume this was set, right?

Reply:
the Id is coming from the props, so it should be set.